components/Banner/src/MDCBannerView.m

```diff
@@ -505,4 +507,54 @@ - (BOOL)isAbleToFitTextLabel:(UILabel *)textLabel withWidthLimit:(CGFloat)widthL
   return size.width <= widthLimit;
 }
 
+#pragma mark - Font
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)mdc_adjustsFontForContentSizeCategory {
+  _mdc_adjustsFontForContentSizeCategory = mdc_adjustsFontForContentSizeCategory;
+
+  if (mdc_adjustsFontForContentSizeCategory) {
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(contentSizeCategoryDidChange:)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+
+  [self updateBannerFont];
+}
+
+- (void)contentSizeCategoryDidChange:(__unused NSNotification *)notification {
+  [self updateBannerFont];
+}
+
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
+  [self updateBannerFont];
+}
+
+- (void)updateBannerFont {
+  [self updateTextFont];
+  [self updateButtonFont];
+
+  [self invalidateIntrinsicContentSize];
+  [self setNeedsUpdateConstraints];
+}
+
+- (void)updateTextFont {
+  UIFont *textFont = self.textLabel.font;
+  if (self.mdc_adjustsFontForContentSizeCategory && textFont.mdc_scalingCurve) {
+    textFont = [textFont mdc_scaledFontForTraitEnvironment:self];
+  }
+  self.textLabel.font = textFont;
+}
+
+- (void)updateButtonFont {
+  self.leadingButton.mdc_adjustsFontForContentSizeCategory =
+      self.mdc_adjustsFontForContentSizeCategory;
+  self.trailingButton.mdc_adjustsFontForContentSizeCategory =
+      self.mdc_adjustsFontForContentSizeCategory;
```

Comment:
Can this method be folded in to the setter for mdc_adjustsFontForContentSizeCategory? I'd expect these values to be kept in sync there.

Reply:
Addressed in f48539c3d7cd07ad36955c1a13b11c89ebecaaa4.